src/java/org/jivesoftware/openfire/net/StanzaHandler.java

```diff
@@ -182,13 +183,28 @@ else if ("compress".equals(tag)) {
                 // resource binding and session establishment (to client sessions only)
                 waitingCompressionACK = true;
             }
+        } else if(isStreamManagementStanza(doc)) {
+        	switch(tag) {
+        		case "enable":
+        			session.enableStreamMangement(doc);
+        			break;
+        		case "r":
+        			session.getStreamManager().sendServerAcknowledgement();
+        			break;
+        		case "a":
+        			session.getStreamManager().processClientAcknowledgement(doc);
+        			break;
+        		default:

```

Comment:
Should probably also return a `<failed><not-implemented/></failed>` (sic) response here for the `<resume />` tag, although admittedly a properly implemented client should not send it since we haven't advertised the capability.